analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -797,6 +795,81 @@ def add_arguments_to_parser(parser):
         func=main, func_process_config_file=cmd_config.process_config_file)
 
 
+def validate_analyzer_parameter(analyzer_conf):
+    """
+    Ensure that the analyzer_config parameter is set to a valid value
+    by verifying if it belongs to the set of allowed values.
+    """
+    analyzer = analyzer_conf.__getitem__(0).analyzer
+
+    if analyzer in analyzer_types.supported_analyzers:
+        analyzer_class = analyzer_types.supported_analyzers[analyzer]
+    else:
+        LOG.error(f"Invalid --analyzer-config parameter: {analyzer} is not "
+                  f"a supported analyzer. The supported analyzers are "
+                  f"{', '.join(a for a in analyzer_types.supported_analyzers)}"
+                  f".")
+        return False
+
+    if not isinstance(analyzer_conf, list):
+        return False
+
+    analyzers = [analyzer[0] for analyzer in
+                 analyzer_class.get_analyzer_config()]
+
+    # Make sure "[clang-tidy] Allow to override checker list #3203" works
+    if analyzer_class.ANALYZER_NAME == 'clang-tidy':
+        analyzers.append('take-config-from-directory')
+
+    for cfg in analyzer_conf:
+        if cfg.analyzer == analyzer_class.ANALYZER_NAME and \
+                cfg.option not in analyzers:
+            LOG.error(f"Invalid --analyzer-config parameter: "
+                      f"{analyzer_class.ANALYZER_NAME} has no config named "
+                      f"{cfg.option}. Use the 'CodeChecker analyzers "
+                      f"--analyzer-config {analyzer_class.ANALYZER_NAME}' "
+                      f"command to check available configs.")
+            return False
+        else:
+            return True
```

Comment:
[Don’t use else after a return](https://llvm.org/docs/CodingStandards.html#id41).
```suggestion
        return True
```